### PR TITLE
extract post_logout_redirect_uri from the session

### PR DESCRIPTION
**Pull request by Chamodii**

### CODE REVIEW

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 1]
```diff
@@ -703,6 +703,11 @@ private void handleMissingSessionStateGracefully(HttpServletRequest request, Htt
         String idTokenHint = request.getParameter(OIDCSessionConstants.OIDC_ID_TOKEN_HINT_PARAM);
         String postLogoutRedirectUri = request.getParameter(OIDCSessionConstants.OIDC_POST_LOGOUT_REDIRECT_URI_PARAM);
         if (StringUtils.isEmpty(idTokenHint) || StringUtils.isEmpty(postLogoutRedirectUri)) {
+            if(StringUtils.isNotEmpty(request.getParameter("sessionDataKey"))){
+                postLogoutRedirectUri=
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: format

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 2]
```diff
@@ -703,6 +703,11 @@ private void handleMissingSessionStateGracefully(HttpServletRequest request, Htt
...
+                    getSessionDataFromCache(request.getParameter("sessionDataKey")).getPostLogoutRedirectUri();
+                redirectURL=postLogoutRedirectUri;
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: format

File: components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java [hunk 3]
```diff
@@ -703,6 +703,11 @@ private void handleMissingSessionStateGracefully(HttpServletRequest request, Htt
         String idTokenHint = request.getParameter(OIDCSessionConstants.OIDC_ID_TOKEN_HINT_PARAM);
         String postLogoutRedirectUri = request.getParameter(OIDCSessionConstants.OIDC_POST_LOGOUT_REDIRECT_URI_PARAM);
         if (StringUtils.isEmpty(idTokenHint) || StringUtils.isEmpty(postLogoutRedirectUri)) {
+            if (StringUtils.isNotEmpty(request.getParameter("sessionDataKey"))) {
+                postLogoutRedirectUri =
+                        getSessionDataFromCache(request.getParameter("sessionDataKey")).getPostLogoutRedirectUri();
+                redirectURL = postLogoutRedirectUri;
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCLogoutServlet.java]: check whether postLogoutRedirectUri is not null and empty

**Comment by darshanasbg** [comment]
This is no longer needed with the fix: https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1173

**Pull request closed without merging**